plugin/pkg/shoot/validator/admission.go

```diff
@@ -366,32 +366,49 @@ func (c *validationContext) validateSeedSelectionForHAShoot() error {
 	return nil
 }
 
-func (c *validationContext) validateScheduling(ctx context.Context, a admission.Attributes, authorizer authorizer.Authorizer, shootLister corelisters.ShootLister) error {
-	if a.GetOperation() == admission.Delete {
-		return nil
-	}
-
-	// If there is already a seedName in the shoot yaml, the scheduler will not create the Binding
-	// So we need to keep the validation here.
+func (c *validationContext) validateScheduling(ctx context.Context, a admission.Attributes, authorizer authorizer.Authorizer, shootLister corelisters.ShootLister, seedLister corelisters.SeedLister, shootStateLister corev1alpha1listers.ShootStateLister) error {
 	var (
 		shootIsBeingScheduled          = c.oldShoot.Spec.SeedName == nil && c.shoot.Spec.SeedName != nil
 		shootIsBeingRescheduled        = c.oldShoot.Spec.SeedName != nil && c.shoot.Spec.SeedName != nil && *c.shoot.Spec.SeedName != *c.oldShoot.Spec.SeedName
 		mustCheckSchedulingConstraints = shootIsBeingScheduled || shootIsBeingRescheduled
 	)
 
-	if shootIsBeingScheduled {
-		if a.GetOperation() == admission.Create {
+	switch a.GetOperation() {
+	case admission.Create:
+		if shootIsBeingScheduled {
 			if err := authorize(ctx, a, authorizer, "set .spec.seedName"); err != nil {
 				return err
 			}
 		}
+	case admission.Update:
+		if a.GetSubresource() == "binding" {
+			if c.shoot.DeletionTimestamp != nil {
```

Comment:
When the shoot has a `DeletionTimestamp` the `spec` field is already immutable independent if the binding subresource is used. Do we need this additional check here?